core/src/main/java/org/apache/gravitino/storage/relational/service/MetadataObjectService.java

```diff
@@ -214,4 +233,407 @@ public static String getMetadataObjectFullName(String type, long metadataObjectI
+        });
+
+    return tableIdAndNameMap;
+  }
+
+  /**
+   * Retrieves a map of Topic object IDs to their full names.
+   *
+   * @param ids A list of Topic object IDs to fetch names for.
+   * @return A Map where the key is the Topic ID and the value is the Topic full name. The map may
+   *     contain null values for the names if its parent object is deleted. Returns an empty map if
+   *     no Topic objects are found for the given IDs. {@code @example} value of topic full name:
+   *     "catalog1.schema1.topic1"
+   */
+  public static Map<Long, String> getTopicObjectsFullName(List<Long> ids) {
+    List<TopicPO> topicPOs =
+        SessionUtils.getWithoutCommit(
+            TopicMetaMapper.class, mapper -> mapper.listTopicPOsByTopicIds(ids));
+
+    if (topicPOs == null || topicPOs.isEmpty()) {
+      return new HashMap<>();
+    }
+
+    List<Long> catalogIds =
+        topicPOs.stream().map(TopicPO::getCatalogId).collect(Collectors.toList());
+    List<Long> schemaIds = topicPOs.stream().map(TopicPO::getSchemaId).collect(Collectors.toList());
+
+    Map<Long, String> catalogIdAndNameMap = getCatalogIdAndNameMap(catalogIds);
+    Map<Long, String> schemaIdAndNameMap = getSchemaIdAndNameMap(schemaIds);
+
+    HashMap<Long, String> topicIdAndNameMap = new HashMap<>();
+
+    topicPOs.forEach(
+        topicPO -> {
+          // since the catalog or schema can be deleted, we need to check the null value,
+          // and when catalog or schema is deleted, we will set fullName of topicPO to
+          // null
+          String catalogName = catalogIdAndNameMap.getOrDefault(topicPO.getCatalogId(), null);
+          if (catalogName == null) {
+            LOG.warn("The catalog of topic {} may be deleted", topicPO.getTopicId());
+            topicIdAndNameMap.put(topicPO.getTopicId(), null);
+            return;
+          }
+
+          String schemaName = schemaIdAndNameMap.getOrDefault(topicPO.getSchemaId(), null);
+          if (schemaName == null) {
+            LOG.warn("The schema of topic {} may be deleted", topicPO.getTopicId());
+            topicIdAndNameMap.put(topicPO.getTopicId(), null);
+            return;
+          }
+
+          String fullName = DOT_JOINER.join(catalogName, schemaName, topicPO.getTopicName());
+          topicIdAndNameMap.put(topicPO.getTopicId(), fullName);
+        });
+
+    return topicIdAndNameMap;
+  }
+
+  /**
+   * Retrieves a map of Column object IDs to their full names.
+   *
+   * @param ids A list of Column object IDs to fetch names for.
+   * @return A Map where the key is the Column ID and the value is the Column full name. The map may
+   *     contain null values for the names if its parent object is deleted. Returns an empty map if
+   *     no Column objects are found for the given IDs. {@code @example} value of column full name:
+   *     "catalog1.schema1.table1.column1"
+   */
+  public static Map<Long, String> getColumnObjectsFullName(List<Long> ids) {
```

Comment:
This method is not used, so your plan is?

Reply:
I thought `Column` is needed before.

I've removed column related code in this PR.